stl/inc/algorithm

```diff
@@ -2671,6 +2671,51 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // ALIAS TEMPLATE swap_ranges_result
+    template <class _In1, class _In2>
+    using swap_ranges_result = in_in_result<_In1, _In2>;
+
+    // VARIABLE ranges::swap_ranges
+    class _Swap_ranges_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It1, sentinel_for<_It1> _Se1, input_iterator _It2, sentinel_for<_It2> _Se2>
+            requires indirectly_swappable<_It1, _It2>
+        constexpr swap_ranges_result<_It1, _It2>
+        operator()(_It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2) const {
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+            auto _UFirst1 = _Get_unwrapped(_STD move(_First1));
+            auto _ULast1 = _Get_unwrapped(_STD move(_Last1));
+            auto _UFirst2 = _Get_unwrapped(_STD move(_First2));
+            auto _ULast2 = _Get_unwrapped(_STD move(_Last2));
+            while ((_UFirst1 != _ULast1) && (_UFirst2 != _ULast2)) {

```

Comment:
Our codebase's conventions don't add parentheses around comparisons here, as they have higher precedence than logical operators. (We parenthesize everything around bitwise operators which are notoriously confusing, and we also add parentheses to `(X && Y) || Z` because compilers like to complain otherwise.)